cypress/integration/regression/decision-tree-regression.spec.ts

```diff
@@ -0,0 +1,488 @@
+/* eslint-disable no-undef */
+import { datasetPage } from '../../page-objects/app/datasets-page'
+import { decisionTreesPage } from '../../page-objects/app/decision-trees-page'
+
+describe('Regression test of the decision tree', () => {
+  const includedVariants = '+5,041,176'
+  const datasetName = 'xl_PGP3140_wgs_NIST-4_2'
+
+  const text =
+    ' \nif Compound_Request(request = [[1,  {"0": ["HG003", "HG004"], "1-2": ["HG002"]} ]]) in   :\n     return true'
+
+  const inheritanceMode = 'Inheritance_Mode'
+  const selectAll = 'Select All'
+  const compoundRequest = 'Compound_Request'
+
+  it('should open decision tree for XL dataset', () => {
+    datasetPage.visit()
+    datasetPage.leftPanel.leftPanelHeader.haveText('Datasets')
+    datasetPage.leftPanel.datasetsListElem.getButtonByText(datasetName).click()
+    datasetPage.datasetInfo.datasetHeader.haveText(datasetName)
+    datasetPage.datasetInfo.openInViewer.click()
+    datasetPage.datasetInfo.viewerOption.contains('Decision Tree Panel').click()
+    cy.url().should('include', `/filter?ds=${datasetName}`)
+  })
+  it('should search attribute based on a substring', () => {
+    decisionTreesPage.visit(`/filter?ds=${datasetName}`)
+    decisionTreesPage.decisionTreeResults.addAttribute.click()
+    decisionTreesPage.attributesList.searchForAttr.eq(0).type('aller')
+    decisionTreesPage.decisionTreeResults.graphHeaders
+      .eq(0)
+      .should('have.text', 'Callers')
+  })
+
+  it('should select all attributes | step 3', () => {
+    decisionTreesPage.visit(`/filter?ds=${datasetName}`)
+    decisionTreesPage.decisionTreeResults.addAttribute.click()
+    decisionTreesPage.attributesList.searchForAttr.eq(0).type('aller')
+    decisionTreesPage.decisionTreeResults.graphHeaders.eq(0).click()
+    cy.waitUntil(() =>
+      decisionTreesPage.attributesList.selectAll.element.then(el => {
+        return el.text() === selectAll
+      }),
+    )
+    decisionTreesPage.attributesList.selectAll.contains(selectAll).click()
+    cy.intercept('POST', '/app/statunits').as('applyAttributes')
+    decisionTreesPage.attributesList.addSelectedAttributes.click()
+    cy.wait('@applyAttributes').its('response.statusCode').should('eq', 200)
+    decisionTreesPage.decisionTreeResults.excludeInfo
+      .first()
+      .should('have.text', includedVariants)
+  })
+
+  it('should join second attribute with first | step 4', () => {
+    decisionTreesPage.visit(`/filter?ds=${datasetName}`)
```

Comment:
I see the similar steps in your tests so you can create actions in decisionTreesPage and use them in your tests.